core/src/test/scala/integration/kafka/server/KRaftClusterTest.scala

```diff
@@ -804,28 +804,24 @@ class KRaftClusterTest {
         val quorumState = admin.describeMetadataQuorum(new DescribeMetadataQuorumOptions)
         val quorumInfo = quorumState.quorumInfo.get()
 
-        assertEquals(4, quorumInfo.observers.size)
-        assertEquals(3, quorumInfo.voters.size)
+        assertEquals(cluster.controllers.asScala.keySet, quorumInfo.voters.asScala.map(_.replicaId).toSet)
         assertTrue(2999 < quorumInfo.leaderId && 3003 > quorumInfo.leaderId,
```

Comment:
We could make this assertion a little more generic by asserting that `quorumInfo.leaderId` is contained in `cluster.controllers.asScala.keySet`.